Update color scheme & initialize CSS variables in theme

pascalwengerter opened the pull request:
**Update:** Successfully replaced all SASS variables with custom CSS properties and the design system _mostly_ works as expected again. We'll have to double-check all components to reveal hidden `uikit` styles (and integrate those).

pascalwengerter commented:
Latest changes introduce CSS variables to almost all components, add new variants & more themable colors. Still WIP

pascalwengerter commented:
Successfully rebased this on the recent changes in `master`

pascalwengerter commented:
@JammingBen @JanAckermann could you check this out locally and give it a review? (Not everything is perfect yet, tests need to be updated and e.g. `ocbutton.scss` needs to be restructured)

JanAckermann commented:
Massive change, but looks good to me so far :rocket:

JammingBen commented on src/components/OcButton.vue:
```diff
@@ -84,13 +84,21 @@ export default {
+        return value.match(/(passive|primary|success|danger|warning)/)
+      },
+    },
+    appearance: {
```
Docstring to keep it consistent. And developer-friendly 🙂

pascalwengerter replied:
Great catch, thanks!

JammingBen commented on src/components/OcModal.vue:
```diff
@@ -13,28 +13,30 @@
           key="modal-input"
           ref="ocModalInput"
-          v-model="$_ocModal_input_value"
+          v-model="input_value"
```
`inputValue` or even `input`?

The same goes for some more properties in this component. I get the intention of a prefix for the input related stuff, but it is already kinda mixed up all over the place... So I vote for omitting the prefix, or at least keep it camelCase :)

kulmann replied:
`inputValue` (camel case please :-) )

pascalwengerter replied:
Should be resolved now!

JammingBen reviewed (commented):
Awesome! Just had a look at the code, not yet the ODS.

You didn't rebase with master, did you? Because some of the recent commits are listed as change 🤔  
Edit: ahhh okay, nvm, just noticed the PR goes into `configurable-design-tokens-in-ODS`.

pascalwengerter commented:
> Awesome! Just had a look at the code, not yet the ODS.
> 
> You didn't rebase with master, did you? Because some of the recent commits are listed as change 🤔
> Edit: ahhh okay, nvm, just noticed the PR goes into `configurable-design-tokens-in-ODS`.

Yeah sorry, I forgot the fact that I'm targeting another branch (to collect changes there and not block `master`) but forgot to rebase that one too, which has just been done. Thanks for the comments, will work them in later! 🙏🏽

kulmann commented on src/components/OcModal.vue:
```diff
@@ -1,8 +1,8 @@
+    <div ref="$_ocModal" :class="classes" tabindex="0">
       <div class="oc-modal-title">
-        <oc-icon v-if="icon" :name="icon" :variation="$_ocModal_titleIcon_variation" />
+        <oc-icon v-if="icon" :name="icon" :variation="this.variation" />
```
```suggestion
        <oc-icon v-if="icon" :name="icon" :variation="variation" />
```

pascalwengerter commented:
Updates this PR, mostly with following changes:

- added mixins for alert & button components
- removed scoped styles
- removed unused visibility helper class
- reintroduced muted background to modal body
- reintroduced inverse icon
- reintroduced filled button in searchbar
- reintroduced raw buttons for acceptance/declare of shares
- updated modal to make buttons customizable
- updated snapshot tests

kulmann commented on src/styles/theme/oc-progress.scss:
```diff
@@ -59,17 +52,59 @@ $progress-height-small: 5px !default;
   }
 
+  &-primary {
+    &::-webkit-progress-value {
```
Can be collapsed to less lines by listing the class names comma separated:

```
&::-webkit-progress-value,
&::-moz-progress-bar,
&::-ms-fill {
  background-color: var(--oc-variation-primary);
}
```

Same for the other variations.

kulmann commented on src/styles/theme/oc-radio.scss:
```diff
@@ -12,13 +7,17 @@
 @mixin hook-form-radio {
```
I think we don't need this mixin

kulmann commented on src/styles/theme/oc-radio.scss:
```diff
@@ -12,13 +7,17 @@
 
 .oc-radio {
   @extend .uk-radio;
 
```
Could you change the default style so that the radiobutton uses the same input background like the checkbox? Should be solved by moving the background-color and border from the `hook-form-radio` mixin above to the `.oc-radio` class:

```
background-color: var(--oc-input-background);
border: 1px solid var(--oc-input-border);
```

kulmann replied:
Also, could you add the same focus style that the checkbox already has? It's not necessarily in scope for this PR, but a quick win.

kulmann commented on docs/components/tokens/Icon.vue:
```diff
@@ -1,9 +1,8 @@
+      <oc-icon :name="filename" size="large" variation="primary"></oc-icon>
       <oc-icon :name="filename" size="large" variation="success"></oc-icon>
       <oc-icon :name="filename" size="large" variation="warning"></oc-icon>
       <oc-icon :name="filename" size="large" variation="danger"></oc-icon>
```
As discussed, please add the `inverse` variation for the example

kulmann commented on src/components/OcModal.vue:
```diff
@@ -13,28 +13,31 @@
           <oc-button
             class="oc-modal-body-actions-confirm oc-ml-s"
-            :variation="$_ocModal_buttonConfirm_variation"
+            :variation="buttonConfirmVariation"
```
```suggestion
            :variation="buttonConfirmVariation || variation"
```

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +114,53 @@ export default {
-     * Text of confirm button
+     * Variation type of the cancel button
+     */
+    buttonCancelVariation: {
```
Add validation to this prop, same as validation of the `variation` prop

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +114,53 @@ export default {
+    /**
+     * Appearance of the cancel button
+     */
+    buttonCancelAppearance: {
```
add validation to this prop

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +114,53 @@ export default {
+    /**
+     * Appearance of the confirm button
+     */
+    buttonConfirmAppearance: {
```
Add validation to this prop

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +114,53 @@ export default {
+    buttonConfirmVariation: {
+      type: String,
+      required: false,
+      default: "passive",
```
```suggestion
      default: null,
```

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +114,53 @@ export default {
+    /**
+     * Variation type of the confirm button
+     */
+    buttonConfirmVariation: {
```
add validation to this prop, but also allow `null`, should work like this:

```js
validator: value => {
  return value === null || value.match(/(passive|primary|danger|success|warning)/)
},
```

kulmann reviewed (changes requested):
Awesome piece of work, I love it! 🚀 
Added some small things that need fixing/improvement, most of them already discussed when we reviewed together. Looking forward to having this merged and bringing it to life in oC Web!

kulmann commented on src/components/OcModal.vue:
```diff
@@ -106,21 +115,65 @@ export default {
+      required: false,
+      default: null,
+      validator: value => {
+        return value.match(/(passive|primary|danger|success|warning)/) || value === null
```
Has to be
```suggestion
        return value === null || value.match(/(passive|primary|danger|success|warning)/)
```
otherwise `value.match` will fail hard when `value` evaluates to `null`.

kulmann reviewed (approved):
LGTM now! 🚀
